corehq/messaging/scheduling/views.py

```diff
@@ -992,3 +1003,102 @@ def dispatch(self, request, *args, **kwargs):
                       "properties in this alert.")
                 )
             return super(EditConditionalAlertView, self).dispatch(request, *args, **kwargs)
+
+
+class DownloadConditionalAlertView(ConditionalAlertBaseView):
+    urlname = 'download_conditional_alert'
+    http_method_names = ['get']
+
+    @method_decorator(toggles.BULK_CONDITIONAL_ALERTS.required_decorator())
+    def dispatch(self, *args, **kwargs):
+        return super(DownloadConditionalAlertView, self).dispatch(*args, **kwargs)
+
+    def get(self, request, domain):
+        title = _("Conditional Alerts")
+        headers = ((title, (_('id'), _('name'), _('case_type'))),)
+        rows = [(
+            rule.pk,
+            rule.name,
+            rule.case_type,
+        ) for rule in self.get_conditional_alerts_queryset()]
+
+        temp = io.BytesIO()
+        export_raw(headers, [(title, rows)], temp)
+        filename = '{title} - {domain}'.format(
+            domain=domain,
+            title=title)
+        return export_response(temp, Format.XLS_2007, filename)
+
+
+class UploadConditionalAlertView(BaseMessagingSectionView):
+    urlname = 'upload_conditional_alert'
+    page_title = ugettext_lazy("Upload Conditional Alerts")
+    template_name = 'hqwebapp/bulk_upload.html'
+
+    @method_decorator(toggles.BULK_CONDITIONAL_ALERTS.required_decorator())
+    @method_decorator(reminders_framework_permission)
+    def dispatch(self, *args, **kwargs):
+        return super(UploadConditionalAlertView, self).dispatch(*args, **kwargs)
+
+    @property
+    def page_context(self):
+        context = {
+            'bulk_upload': {
+                "download_url": reverse("download_conditional_alert", args=(self.domain,)),
+                "adjective": _("conditional alert"),
+                "plural_noun": _("conditional alerts"),
+            },
+        }
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(context),
+        })
+        return context
+
+    @property
+    def parent_pages(self):
+        return [{
+            'title': BroadcastListView.page_title,
+            'url': reverse(BroadcastListView.urlname, args=[self.domain]),
+        }]
+
+    def post(self, request, *args, **kwargs):
+        try:
+            worksheet = get_single_worksheet(request.FILES['bulk_upload_file'])
+        except WorkbookJSONError as e:
+            messages.error(request, six.text_type(e))
+            return self.get(request, *args, **kwargs)
+
+        success_count = 0
+        errors = []
+        for index, row in enumerate(worksheet):
+            index = index + 2   # one-indexed, plus header row
```

Comment:
nit: `enumerate` has a `start` parameter which you could set to `2` (https://docs.python.org/2/library/functions.html#enumerate)